elastic/beats
# Initial work toward synthetic monitors

Initial, experimental version of synthetic monitor support for Heartbeat. This works in concert with https://github.com/elastic/synthetics , and is intended to only be run within the docker container built in the synthetics repo. This is guarded by the new `ELASTIC_SYNTHETICS_CAPABLE` env variable which must be set to `true`. As a note, this is easy to bypass because the main purpose is to make it clear that we do not support synthetics outside of the docker container for now.

This includes a new directory, `sample-synthetics-config` in the x-pack directory that demonstrates using heartbeat with synthetics. In the future 

## Checklist


## How to test this PR locally

You can check that this PR works by running (in the x-pack/heartbeat directory): 
`mage build && env ELASTIC_SYNTHETICS_CAPABLE=true ./heartbeat --path.config sample-synthetics-config/ -e`
